modal/mount.py

```diff
@@ -558,6 +567,21 @@ async def lookup(
         await resolver.load(obj)
         return obj
 
+    async def _deploy(
+        self: "_Mount",
+        deployment_name: Optional[str] = None,
+        namespace=api_pb2.DEPLOYMENT_NAMESPACE_WORKSPACE,
+        environment_name: Optional[str] = None,
+        client: Optional[_Client] = None,
+    ) -> "_Mount":
+        self._deployment_name = deployment_name
+        self._namespace = namespace
+        self._environment_name = environment_name
+        if client is None:
+            client = await _Client.from_env()
+        resolver = Resolver(client=client)
+        await resolver.load(self)
```

Comment:
I'm a little confused by this; does deployment happen as a side-effect of calling `Resolver.load` on an object??

Reply:
Yeah in this case – `Resolver.load` is the way you trigger an object's initialization code. It's a bit confusing.